Homeworks/Homework W8/SQLQuery_HomeworkW8.sql

```diff
@@ -0,0 +1,340 @@
+SELECT z.[Zone], w.FirstName
+FROM ZoneArea z
+JOIN Workers  w ON z.Worker=w.Id;
+
+SELECT * 
+FROM Ticket;
+
+INSERT INTO Ticket
+VALUES(10,'2023-01-20','2023-01-20',1);
+INSERT INTO Ticket
+VALUES(10,'2023-01-20','2023-01-20',2);
+INSERT INTO Ticket
+VALUES(50,'2023-01-20','2023-01-27',3);
+INSERT INTO Ticket
+VALUES(80,'2023-01-20','2023-02-20',2);
+INSERT INTO Ticket
+VALUES(50,'2023-02-10','2023-02-20',2,4);
+INSERT INTO Ticket
+VALUES(10,'2023-01-22','2023-01-22',4,5);
+INSERT INTO Ticket
+VALUES(80,'2023-01-24','2023-02-24',5,6);
+
+ALTER TABLE Visitors
+DROP constraint FK__Visitors__Ticket__44FF419A;
+ALTER TABLE Visitors
+DROP COLUMN Ticket;
+
+ALTER TABLE Ticket
+ADD Visitor INT REFERENCES Visitors(Id);
+
+SELECT * 
+FROM Ticket;
+
+UPDATE Ticket
+SET Visitor = 3 
+WHERE Id = 9;
+
+SELECT *
+FROM Visitors;
+SELECT * 
+FROM Ticket;
+
+INSERT INTO Visitors
+VALUES('Alexandru','Mihai');
+INSERT INTO Visitors
+VALUES('Loredana','Luca');
+INSERT INTO Visitors
+VALUES('Elvis','Musteata');
+INSERT INTO Visitors
+VALUES('Emanuela','Coroi');
+INSERT INTO Visitors
+VALUES('George','Buhusi');
+INSERT INTO Visitors
+VALUES('Gina','Doroh');
+
+INSERT INTO Animals
+VALUES (1);
+INSERT INTO Animals
+VALUES (1);
+INSERT INTO Animals
+VALUES (1);
+INSERT INTO Animals
+VALUES (1);
+INSERT INTO Animals
+VALUES (2);
+INSERT INTO Animals
+VALUES (2);
+INSERT INTO Animals
+VALUES (2);
+INSERT INTO Animals
+VALUES (2);
+INSERT INTO Animals
+VALUES (2);
+INSERT INTO Animals
+VALUES (3);
+INSERT INTO Animals
+VALUES (4);
+INSERT INTO Animals
+VALUES (5);
+INSERT INTO Animals
+VALUES (6);
+INSERT INTO Animals
+VALUES (6);
+INSERT INTO Animals
+VALUES (7);
+INSERT INTO Animals
+VALUES (7);
+--
+INSERT INTO Animals
+VALUES (8);
+INSERT INTO Animals
+VALUES (8);
+INSERT INTO Animals
+VALUES (9);
+INSERT INTO Animals
+VALUES (9);
+INSERT INTO Animals
+VALUES (10);
+INSERT INTO Animals
+VALUES (11);
+INSERT INTO Animals
+VALUES (12);
+INSERT INTO Animals
+VALUES (12);
+INSERT INTO Animals
+VALUES (13);
+INSERT INTO Animals
+VALUES (14);
+INSERT INTO Animals
+VALUES (15);
+INSERT INTO Animals
+VALUES (16);
+INSERT INTO Animals
+VALUES (17);
+INSERT INTO Animals
+VALUES (17);
+INSERT INTO Animals
+VALUES (18);
+INSERT INTO Animals
+VALUES (18);
+INSERT INTO Animals
+VALUES (19);
+INSERT INTO Animals
+VALUES (19);
+INSERT INTO Animals
+VALUES (20);
+INSERT INTO Animals
+VALUES (20);
+INSERT INTO Animals
+VALUES (21);
+INSERT INTO Animals
+VALUES (21);
+INSERT INTO Animals
+VALUES (21);
+INSERT INTO Animals
+VALUES (20);
+INSERT INTO Animals
+VALUES (18);
+INSERT INTO Animals
+VALUES (17);
+INSERT INTO Animals
+VALUES (16);
+INSERT INTO Animals
+VALUES (15);
+
+SELECT *
+FROM Animals;
+SELECT *
+FROM Species;
+SELECT *
+FROM Ticket;
+SELECT *
+FROM Visitors;
+SELECT *
+FROM Workers;
+SELECt *
+FROM ZoneArea;
+-- Requirements
+
+--the number of animals as a total and in each species
+
+SELECT COUNT( a.Id) as TotalOfAnimals
```

Comment:
I like that you have more exercises than requested. It shows that you want to practice. Great job!